crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -72,9 +72,34 @@ async fn create(
         }
     };
 
-    let new_dimension = Dimension {
+    let req_priority_val: i32 = create_req.priority.clone().into();
+    let dimension_name: String = create_req.dimension.clone().into();
+    let req_position_val = Into::<Option<i32>>::into(create_req.position.clone());
+
+    let mut results: Vec<(String, i32, i32)> = dimensions
+        .order(priority.asc())
+        .select((dimension, priority, position))
+        .load::<(String, i32, i32)>(&mut conn)
+        .map_err(|err| {
+            log::error!("failed to fetch dimensions with error: {}", err);
+            unexpected_error!("Something went wrong")
+        })?;
```

Comment:
Should we throw a DB error. Also
```suggestion
        .map_err(|err| {
            log::error!("failed to fetch dimensions with error: {}", err);
            unexpected_error!("Could not process this request due to a database error. Please reach out to your admin")
        })?;
```